webpack/generateDetails.js

```diff
@@ -0,0 +1,168 @@
+/* eslint-env node */
+/* eslint-disable no-useless-escape */
+// NOTE: double escapes are needed for file path delimiters in webpack regexes for cross-platform support
+
+const fs = require('fs');
+const path = require('path');
+const appPaths = require('./appPaths');
+
+const scanTemplates = function (dir, entryRegex, allAppPaths, details, isProdMode) {
+    /**
+     * This recursively scans all template files within a given directory, looking
+     * for a metch to the `entryRegex` specified above. The first group this entry regex
+     * (`match[1]`) should return the webpack entry name, eg `hqwebapp/js/some_page`.
+     *
+     * As they are discovered, entries are appended to the `details.entries` dictionary.
+     * Additionally, `details.aliases` is updated when an app alias is first found,
+     * eg `hqwebapp/js` (the first two folders in the entry name).
+     * Lastly, the (unique) application name is added to `details.appsWithEntries`
+     *
+     * @param dir - path to directory to begin scanning
+     * @param entryRegex -  regex for identifying an entry in an .html page (see examples below)
+     * @param allAppPaths (dict) - keys are all the applications being scanned
+     *                             and values are paths to each application's folder
+     * @param details (dict) - the dictionary that will be modified by this function, formatted as follows:
+     *                          {
+     *                              entries: {},
+     *                              aliases: {},
+     *                              appsWithEntries: [],
+     *                          }
+     * @param isProdMode (boolean) - `true` if this should be run in production mode, the difference
```

Comment:
@biyeun Can you explain more about this? My current understanding (likely incomplete) is that even locally, developers will be running the webpack build process and re-generating webpack bundles on every change to a js file. What's the advantage/disadvantage of including the content hashes in the bundle file names for local development?

Also, will `isProdMode` be true or false on staging?